### DISCUSSION TITLE
How to get another sensor state in JS

### DISCUSSION THREAD

**Discussion by WallK** [ExperienceLovelace/ha-floorplan#399] [Q&A] [answered]
Hello
I'm trying to do style_set using states of an entity and another sensor's state to calculate the color

Basically, I want to change temperature color using temperature value and humidity value in the same room

It's set up like this:
```
entities:
	- sensor.mitemp_a_temperature
	- sensor.mitemp_b_temperature
	- sensor.mitemp_c_temperature
state_action:
	- service: floorplan.text_set
	  service_data: ${entity.state + " °C"}
	- service: floorplan.style_set
	  service_data:
	    style: <here goes the code>
```

I need to calculate value for sensor.mitemp_a_temperature using sensor.mitemp_a_humidity, sensor.mitemp_b_temperature using sensor.mitemp_b_humidity and so on

How do I get the state of sensor.mitemp_a_humidity inside the JS code?

Thanks!

**Comment by OzGav** [comment, marked as answer]
All entities are exposed. Use the hass entity. For example.

`service_data: '${parseFloat(entity.state.replace(/\s{2,}/g,"") / hass.states["sensor.shellyplus1pm_a8032abc17c4_switch_0_voltage"].state).toFixed(1)}A'`

**Reply by WallK** [reply]
Oh, didn't know about "hass" object!
This will do, thank you!

**Reply by exetico** [reply]
I'm happy that you figured it out, with some help. I'll recommend you to take your time and read through our docs:
https://experiencelovelace.github.io/ha-floorplan/docs/usage/#services

Here the hass object are mentioned too, but would use some more "in-depth" info, for sure 😄 . Theres's a bit of info here: https://developers.home-assistant.io/docs/dev_101_hass/ (which are also mentioned in the docs)

**Reply by WallK** [reply]
I somehow missed the shortest word on a page haha (: 
My floorplan growing more and more to look like an application all in itself, thank you!
